File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program109.java
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and accept one character. 
//		Return index of last occurrence of that character.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


import java.util.Scanner;

class Solution
{
	String str;
	int iCnt;
	char ch;

	Solution(String str,char ch)
	{
		this.str=str;
		this.iCnt=0;
		this.ch=ch;
	}

	int LastChar()
	{
		iCnt=str.length()-1;
		
		while(iCnt!=-1)
		{
			if(str.charAt(iCnt)==ch)
			{
				break;
			}
			iCnt--;
		}
		
		return iCnt;
	}
}

class Program
{
	public static void main(String args[])
	{
		int iRet=0;
		Scanner sObj = new Scanner(System.in);
		System.out.println("Enter the string: ");
		String str = sObj.nextLine();

		System.out.println("Enter the character: ");
		char ch = sObj.next().charAt(0);

		Solution obj = new Solution(str,ch);
		iRet = obj.LastChar();

		System.out.println("Last Occurance is: "+iRet);
	}
}

/*
OUTPUT:
Enter the string: 
Vishal Patil
Enter the character: 
a
Last Occurance is: 9

*/

File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program109.cpp
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept string from user and accept one character. 
//		Return index of last occurrence of that character.
//
/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

class Solution
{

	private:
		string str;
		int iCnt,iLast=0;
		char ch;

	public:

		Solution(string str,char ch,int iCnt=0)
		{
			this->str=str;
			this->ch=ch;
		}

		int LastChar()
		{
			iCnt=str.length();

			while(iCnt!=-1)
			{
				if(str[iCnt]==ch)
				{
					break;
				}

				iCnt--;
			
			}

			return iCnt;
		}
};

int main()
{
	string str;
	char ch='\0';
	int iRet=0;
	
	cout<<"Enter string: ";
	getline(cin,str);

	cout<<"Enter character: ";
	cin>>ch;
	
	Solution *sObj = new Solution(str,ch);
	iRet = sObj->LastChar();
	
	cout<<"Last Occurance is: "<<iRet<<"\n";

	delete sObj;
	return 0;

}

/*
OUTPUT:
Enter string: Marvellous Multi OS
Enter character: W
Last Occurance is: -1

Enter string: MArvellous Multi OS
Enter character: M
Last Occurance is: 11
*/
